src/mbgl/util/mapbox.cpp

```diff
@@ -6,53 +6,80 @@ namespace mbgl {
 namespace util {
 namespace mapbox {
 
-const std::string mapbox = "mapbox://";
+const std::string protocol = "mapbox://";
+const std::string baseURL = "https://api.mapbox.com/";
+const std::string draft = "draft";
+
+struct MapboxURLParts {
+    std::string endpoint;
+    std::string user;
+    std::string id;
+    bool draft;
+};
+
+bool isMapboxURL(const std::string& url) {
+    return url.compare(0, protocol.length(), protocol) == 0;
+}
+
+void parseMapboxURL(const std::string& url, MapboxURLParts &parts) {
+    std::size_t endpointEndIndex = url.find('/', protocol.length());
+    std::size_t userStartIndex = endpointEndIndex + 1;
+    std::size_t userEndIndex = url.find('/', userStartIndex);
+    std::size_t idStartIndex = userEndIndex + 1;
+    std::size_t idEndIndex = url.find('/', idStartIndex);
 
-std::string normalizeURL(const std::string& url, const std::string& pathPrefix, const std::string& accessToken) {
-    if (accessToken.empty())
-        throw std::runtime_error("You must provide a Mapbox API access token for Mapbox tile sources");
+    parts.endpoint = url.substr(protocol.length(), endpointEndIndex - protocol.length());
+    parts.user = url.substr(userStartIndex, userEndIndex - userStartIndex);
+    parts.id = url.substr(idStartIndex, idEndIndex - idStartIndex);
 
-    return std::string("https://api.tiles.mapbox.com")
-        + pathPrefix
-        + url.substr(mapbox.length())
-        + "?access_token="
-        + accessToken;
+    parts.draft = url.compare(idEndIndex + 1, draft.length(), draft) == 0;
 }
 
+
 std::string normalizeSourceURL(const std::string& url, const std::string& accessToken) {
-    if (url.compare(0, mapbox.length(), mapbox) != 0)
+    if (!isMapboxURL(url)) {
         return url;
-
-    std::string result = normalizeURL(url + ".json", "/v4/", accessToken);
-
-    // TileJSON requests need a secure flag appended to their URLs so
-    // that the server knows to send SSL-ified resource references.
-    result += "&secure";
-
-    return result;
+    } else {
```

Comment:
Don't nest `else`s after a guard clause and early return. The reason for guard clauses is to get the uninteresting case taken care of and off your mental stack when reading a method -- using an `else` and extra level of nesting defeats that.